woocommerce/woocommerce-android
# Issue/12414 custom fields editor

Closes: #12414


### Description
This PR is the first part in implementing the custom fields editor, it adds a new screen to edit existing custom fields, the same screen will be used when adding new fields too, and that should be taken into consideration when reviewing the code.

Many parts are still missing to allow keeping the PR size manageable, this includes the following:
1. Saving changes.
2. Returning results to the list screen and updating its state.
3. Using Aztec when editing HTML values.

**While the PR seems large, a big part of the changes is just tests and XML file changes due to using a nested nav graph, so it should be still easy to review.**

### Testing information
1. Open an order in wp-admin.
2. Tap on the "Custom Fields" section is visible (from the screen options at the top of the page)
3. Add some custom fields (for a more comprehensive testing, add fields that contain a URL, an email and a phone number)
5. Open the order in the app.
6. Tap on "View custom fields"
7. Tap on a custom field.
8. Make some changes.
9. Tap on the back button
10. Confirm a dialog is shown.
11. Test the behavior of the dialog buttons.

### The tests that have been performed
The above steps in addition to the validation rules.

### Images/gif
<img src="https://github.com/user-attachments/assets/a2842e36-e47d-4350-80d7-319182d5aa1f" width=360 />


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
